src/pytrajplot/plotting/plot_map.py

```diff
@@ -100,6 +100,46 @@ def crop_map(
         ) = get_dynamic_zoom_boundary(custom_domain_boundaries, origin_coordinates)
 
     padding = 1  # padding on each side, for the dynamically created plots
+    if domain == "dynamic":
+        # Restrict aspect ratio to avoid overstretched maps
+        min_aspect = 1.8
+        max_aspect = 3
+        domain_aspect = (custom_domain_boundaries[1] - custom_domain_boundaries[0]) / (
+            custom_domain_boundaries[3] - custom_domain_boundaries[2]
+        )
+        if domain_aspect < min_aspect:
+            # Add half of the difference on each side
+            custom_domain_boundaries[0] -= 0.5 * (
+                min_aspect * (custom_domain_boundaries[3] - custom_domain_boundaries[2])
+                - (custom_domain_boundaries[1] - custom_domain_boundaries[0])
+            )
+            custom_domain_boundaries[1] += 0.5 * (
+                min_aspect * (custom_domain_boundaries[3] - custom_domain_boundaries[2])
+                - (custom_domain_boundaries[1] - custom_domain_boundaries[0])
+            )
+        if domain_aspect > max_aspect:
+            # Add half of the difference on each side but keep within [-90 .. 90]
+            custom_domain_boundaries[2] = max(
+                -90 + padding,
+                custom_domain_boundaries[2]
+                - 0.5
+                * (
+                    max_aspect
+                    * (custom_domain_boundaries[1] - custom_domain_boundaries[0])
+                    - (custom_domain_boundaries[3] - custom_domain_boundaries[2])
+                ),
+            )
+            custom_domain_boundaries[3] = min(
+                90 - padding,
+                custom_domain_boundaries[3]
+                + 0.5
+                * (
+                    max_aspect
+                    * (custom_domain_boundaries[1] - custom_domain_boundaries[0])
+                    - (custom_domain_boundaries[3] - custom_domain_boundaries[2])
+                ),
+            )
+
```

Comment:
I found this very hard to read. Maybe it is the formatter (black) doing this, maybe not. Here is a suggestion for refactoring by ChatGPT. Maybe black will end up back at your solution, would be interested to see.

```suggestion
if domain == "dynamic":
    # Constants for aspect ratio constraints
    min_aspect_ratio = 1.8
    max_aspect_ratio = 3
    
    # Calculate the current aspect ratio of the domain
    longitude_range = custom_domain_boundaries[1] - custom_domain_boundaries[0]
    latitude_range = custom_domain_boundaries[3] - custom_domain_boundaries[2]
    current_aspect_ratio = longitude_range / latitude_range
    
    # Adjust longitude boundaries if aspect ratio is less than the minimum
    if current_aspect_ratio < min_aspect_ratio:
        required_longitude_range = min_aspect_ratio * latitude_range
        additional_longitude = required_longitude_range - longitude_range
        custom_domain_boundaries[0] -= additional_longitude / 2  # Expand west boundary
        custom_domain_boundaries[1] += additional_longitude / 2  # Expand east boundary
    
    # Adjust latitude boundaries if aspect ratio is greater than the maximum
    if current_aspect_ratio > max_aspect_ratio:
        required_latitude_range = longitude_range / max_aspect_ratio
        additional_latitude = required_latitude_range - latitude_range
        # Ensure adjustments keep latitude within [-90, 90] and add padding if necessary
        custom_domain_boundaries[2] = max(-90 + padding, custom_domain_boundaries[2] - additional_latitude / 2)  # Contract south boundary
        custom_domain_boundaries[3] = min(90 - padding, custom_domain_boundaries[3] + additional_latitude / 2)  # Contract north boundary
```